daemon/license.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"strings"
+
+	ibclient "github.com/infobloxopen/infoblox-go-client"
+	"github.com/sirupsen/logrus"
+)
+
+//Checks for cloud license in nios
+func CheckForCloudLicense(objMgr *ibclient.ObjectManager) {
+	flag, err := CheckLicense(objMgr, "cloud")
+	if err != nil {
+		logrus.Fatal("error", err)
+	}
+	if !flag {
+		logrus.Fatal("Cloud License not available in Infoblox Appliance. Update and try again..")
+	}
+}
+
+func CheckLicense(objMgr *ibclient.ObjectManager, licenseType string) (flag bool, err error) {
+	license, err := objMgr.GetLicense()
+	if err != nil {
+		return flag, err
+	}
+	for _, v := range license {
+		if strings.ToLower(v.Licensetype) == licenseType {
+			if v.ExpirationStatus != "DELETED" && v.ExpirationStatus != "EXPIRED" {
+				flag = true
```

Comment:
is the flag necessary ? Cant we change it to whatever we discussed on friday ?

Reply:
When testing that was not working, As we discussed, We were checking conditions only if the cloud type is cloud and if there is no cloud type is cloud then we were doing nothing so it was not working as expected

Reply:
still flag isnt necessary.  can you please try the other way i asked ?

Reply:
done